### DISCUSSION TITLE
Error in catkin_make

### DISCUSSION THREAD

**Discussion by saviopullkotil** [fjp/2wd-robot#81] [Q&A]
Hey Franz,
How do I solve this?
In file included from /home/prapti/ros_ws/src/diffbot/diffbot_base/include/diffbot_base/diffbot_hw_interface.h:18,
                 from /home/prapti/ros_ws/src/diffbot/diffbot_base/src/diffbot_hw_interface.cpp:1:
/home/prapti/ros_ws/src/diffbot/diffbot_base/include/diffbot_base/pid.h:6:10: fatal error: diffbot_base/ParametersConfig.h: No such file or directory
    6 | #include <diffbot_base/ParametersConfig.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[2]: * [diffbot/diffbot_base/CMakeFiles/diffbot_hw_interface.dir/build.make:63: diffbot/diffbot_base/CMakeFiles/diffbot_hw_interface.dir/src/diffbot_hw_interface.cpp.o] Error 1
make[1]: * [CMakeFiles/Makefile2:4568: diffbot/diffbot_base/CMakeFiles/diffbot_hw_interface.dir/all] Error 2
make: * [Makefile:141: all] Error 2
Invoking "make -j1 -l1" failed

**Comment by fjp** [comment]
Hey @saviopullkotil, which command did you use to build the workspace? `catkin_make` or catkin tools: `catkin build`?

And which Ubuntu and ROS version are you using?

**Reply by saviopullkotil** [reply]
Hello, Im used catkin_make and im running Ubuntu 20.04 and ROS Noetic

**Reply by fjp** [reply]
Ok that looks good.

Did you follow the [steps to install all source and binary dependencies](https://ros-mobile-robots.com/technical_requirements/#source-dependencies) before running `catkin_make`?

The build of the workspace should definitely work, because the [github action build](https://github.com/ros-mobile-robots/diffbot/actions/runs/4584723018/jobs/8096395756#step:4:1296) succeeded.

**Reply by saviopullkotil** [reply]
I also had a doubt that can I use Arduino UNO and L298N motor driver instead of teensy and I2C driver?

**Reply by fjp** [reply]
Sure, should be possible to use this hardware as well, but it needs some modifications in the microcontroller (mcu) code of the [diffbot_base package (scripts/base_controller)](https://github.com/ros-mobile-robots/diffbot/tree/noetic-devel/diffbot_base/scripts/base_controller/lib). What will be needed specifically is a new motor controller class, [in this folder](https://github.com/ros-mobile-robots/diffbot/tree/noetic-devel/diffbot_base/scripts/base_controller/lib/motor_controllers), that implements the [`motor_controller_interface`](https://github.com/ros-mobile-robots/diffbot/blob/noetic-devel/diffbot_base/scripts/base_controller/lib/motor_controller_interface/motor_controller_interface.h), similar to how it is already done for the [adafruit feature wing](https://github.com/ros-mobile-robots/diffbot/tree/noetic-devel/diffbot_base/scripts/base_controller/lib/motor_controllers), which is one example of a concrete implementation for the `motor_controller_interface`.

But first I recommend you should try to get the catkin workspace to build successfully with the existing software to have a good starting point for rest of the project.

**Reply by fjp** [reply]
Just wanted to confirm that the build worked locally for me on a clean Ubuntu 20.04 with the steps mentioned here in the reamde: https://github.com/ros-mobile-robots/diffbot#installation

One important command is to install the dependencies with:

```
rosdep install --from-paths src --ignore-src -r -y
```

Hope this helps, otherwise please let me know if you encounter the same issue you mention above.
